presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -464,10 +466,30 @@ default void dropView(ConnectorSession session, SchemaTableName viewName)
 
     /**
      * Gets the view data for views that match the specified table prefix.
+     * This optional method may be implemented by connectors that can support fetching
+     * view data in bulk. It is used to implement {@code information_schema.views}.
      */
     default Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, SchemaTablePrefix prefix)
     {
-        return emptyMap();
+        if (prefix.getTable().isPresent()) {
```

Comment:
Since this code is already backwards incompatible, how about we remove the "list one view" part of this api, and change the signature to:
```
default Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, String schema)
```
Then `Metadata` we check `if (prefix.getTable().isPresent()` and if so, we call `getView`, otherwise we call `getViews`

Reply:
That's a good idea, though it needs to be `Optional<String> schema`, since we need to allow listing for all schemas. That's what we do for `listTables()` and `listViews()`.